specs/core/1_custody-game.md

```diff
@@ -0,0 +1,483 @@
+    pubkeys = [state.validator_registry[reveal.revealer_index].pubkey]
+    message_hashes = [hash_tree_root(reveal.period)]
+
+    # Case 2: masked punitive early reveal
+    # Masking prevents proposer stealing the whistleblower reward
+    # Secure under the aggregate extraction infeasibility assumption
+    # See pages 11-12 of https://crypto.stanford.edu/~dabo/pubs/papers/aggreg.pdf
+    if reveal.mask != ZERO_HASH:
+        pubkeys.append(state.validator_registry[reveal.masker_index].pubkey)
+        message_hashes.append(reveal.mask)
+
+    return bls_verify_multiple(
+        pubkeys=pubkeys,
+        message_hashes=message_hashes,
+        signature=reveal.subkey,
+        domain=get_domain(
+            fork=state.fork,
+            epoch=reveal.period * EPOCHS_PER_CUSTODY_PERIOD,
+            domain_type=DOMAIN_CUSTODY_REVEAL,
+        ),
+    )
+```
+
+## Per-block processing
+
+### Transactions
+
+Add the following transactions to the per-block processing, in order the given below and after all other transactions in phase 0.
+
+#### Custody reveals
+
+Verify that `len(block.body.custody_reveals) <= MAX_CUSTODY_REVEALS`.
+
+For each `reveal` in `block.body.custody_reveals`, run the following function:
+
+```python
+def process_custody_reveal(state: BeaconState,
+                           reveal: CustodyReveal) -> None:
+    assert verify_custody_reveal(reveal)
+    revealer = state.validator_registry[reveal.revealer_index]
+
+    # Case 1: non-masked non-punitive non-early reveal
+    if reveal.mask == ZERO_HASH:
+        assert reveal.period == epoch_to_custody_period(revealer.activation_epoch) + revealer.custody_reveal_index
+        # Revealer is active or exited
+        assert is_active_validator(revealer, get_current_epoch(state)) or revealer.exit_epoch > get_current_epoch(state)
+        revealer.custody_reveal_index += 1
+        revealer.max_reveal_lateness = max(revealer.max_reveal_lateness, get_current_period(state) - reveal.period)
+
+    # Case 2: masked punitive early reveal
+    else:
+        assert reveal.period > epoch_to_custody_period(get_current_epoch(state))
+        assert revealer.slashed is False
+        slash_validator(state, reveal.revealer_index, reveal.masker_index)
+        increase_balance(state, reveal.masker_index, base_reward(state, index) // MINOR_REWARD_QUOTIENT)
+
+    proposer_index = get_beacon_proposer_index(state, state.slot)
+    increase_balance(state, proposer_index, base_reward(state, index) // MINOR_REWARD_QUOTIENT)
+
+```
+
+#### Chunk challenges
+
+Verify that `len(block.body.chunk_challenges) <= MAX_CHUNK_CHALLENGES`.
+
+For each `challenge` in `block.body.chunk_challenges`, run the following function:
+
+```python
+def process_chunk_challenge(state: BeaconState,
+                            challenge: ChunkChallenge) -> None:
+    # Verify the attestation
+    assert verify_standalone_attestation(state, convert_to_standalone(state, challenge.attestation))
```

Comment:
From what I can see, `verify_standalone_attestation` doesn't exist nor does `convert_to_standalone`

Reply:
`verify_standalone_attestation` and `convert_to_standalone` are defined [here](https://github.com/ethereum/eth2.0-specs/pull/831/files).